### PR TITLE
Pull Request template

### PR DESCRIPTION
**Pull request by darrikonn**
# Pull Request template

## What
Add a template for pull request descriptions

## Why
This would help reviewers better understand the pull requests purpose

## Screenshots / Gifs
image

## Checklist:

- [x] My changes generate no new warnings
- [x] Linting and formatting pass locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## Checklist:
+
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Linting and formatting pass locally with my changes
```
Comment (eirikurn) [inline .github/pull_request_template.md:25]: I think we should just trust the CI which checks three of these. For some changes I don't bother running all tests, linting, formatting, or builds locally when I can have the CI do it for me.

Comment (darrikonn) [reply .github/pull_request_template.md:25]: Yeah my main wonder was that the review is dismissed when changes are added. In this case if the CI fails on lint/format the reviewer needs to review again (which is fine as long as they don't amend the changes to the last commit and force push).

Comment (darrikonn) [reply .github/pull_request_template.md:25]: Same though for me, can't be bothered to run the tests and linting for every PR. But formatting is a pretty quick task.
Even for this small PR, it failed first due to formatting.

Comment (eirikurn) [reply .github/pull_request_template.md:25]: Right. Let's keep a check about formatting, that's probably the most common fail. We can always add more later.

**Review by arnifreyrs — APPROVED** [review]
awesome, I really like it

**Pull request merged**